### PR TITLE
Updates for PyTorch Lightning 2.0 release

### LINKED ISSUE #264: Failed build on master branch (tests #562)
Workflow failed: [tests #562](https://github.com/drivendataorg/zamba/actions/runs/4457976020)

### LINKED ISSUE #265: Failed build on master branch (tests #563)
Workflow failed: [tests #563](https://github.com/drivendataorg/zamba/actions/runs/4521842991)

### PR DESCRIPTION
**Pull request by ejm714**
### Configure accelerator and devices for Trainer
Updates how we set the accelerator and devices based on recent PyTorch Lightning changes.

[`pl.Trainer`](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.trainer.trainer.Trainer.html#lightning.pytorch.trainer.trainer.Trainer) now accepts `accelerator` ("cpu", "gpu", etc."), `strategy`, and `devices` params instead of `num_gpus`. This PR adds a utility to configure accelerator and devices based on the number of user-specified gpus. 

Pros:
- keeps the command line to the same limited set of options (e.g. just specifying number of gpus instead of exposing both the cpu/gpu option as well as number of devices/cores)
- allows us to validate user-provided input up front to ensure that training won't fail after the long video loading check

Cons
- doesn't allow the user to specify the number of cores; if a GPU is not used, all cores will be used
- doesn't support the new "mps" accelerator for apple silicon

Relevant links:
- [GPU training](https://lightning.ai/docs/pytorch/stable/accelerators/gpu_basic.html)
- [Trainer docs](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.trainer.trainer.Trainer.html#lightning.pytorch.trainer.trainer.Trainer)
- [About strategies](https://lightning.ai/docs/pytorch/stable/extensions/strategy.html)

Closes #264 
Closes #265 

### Other changes based on PTL 2.0 release

The new PTL release makes a bunch of non-backward compatible API changes. This PR also provides fixes for those and sets 2.0 as the floor.

- modifies the syntax for learning rate finding with the separation of `Trainer` and `Tuner` (https://github.com/Lightning-AI/lightning/pull/16462)
- updates epoch end hook names and outputs per https://github.com/Lightning-AI/lightning/pull/16520
- removes python 3.7 which isn't compatible with lightning 2.0

### Bonus fixes
- adds a note in the depth estimation docs about the limitations for multiple individuals in the frame (which came up in a user email)
- only use DDP strategy for multi-gpu; this removes a bunch of unnecessary logging warnings about metric syncing across devices which arose from using DDP on a single GPU; we can take advantage of the new "auto" strategy in PTL 2.0

### CODE REVIEW

**Comment by ejm714** [comment]
@pjbull tests are now passing, this is ready for your review when you have time 🙇

**Review by pjbull — APPROVED** [review]
Looks good to me!
